### PR TITLE
Ticket 29 : edit-product

### PR DESCRIPTION
**Pull request by maximFeldman1**
create put request for update an existing product

### CODE REVIEW

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 1]
```diff
@@ -21,19 +27,42 @@ const Root = styled.div`
...
+      onSuccess: handleSuccess,
+    }
+  );
+
+  const { mutate: editMutate } = useMutation(editProduct, {
+    onSuccess: () => {
+      handleSuccess;
+      onDone;
+    },
   });
```
Comment (amitmarko) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx]: Please send `onSubmit` from outside and remove all those.

```suggestion
```

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 2]
```diff
@@ -6,11 +6,17 @@ import { Button } from "primereact/button";
...
 import { ProductService } from "../../../../services/product-service";
-import { useNavigate } from "react-router-dom";
 import { InputField } from "../../../../components/UI/Forms/InputField";
 import { validationYup } from "../../../../schema/validations/validationSchema";
 import { useTranslation } from "react-i18next";
 
+interface IProps {
+  productData?: IProduct;
+  handleSuccess: () => void;
+  onCancel?: () => void;
```
Comment (amitmarko) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx:12]:
```suggestion
  onCancel: () => void;
```

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 3]
```diff
@@ -6,11 +6,17 @@ import { Button } from "primereact/button";
...
+  onDone?: () => void;
```
Comment (amitmarko) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx:13]:
```suggestion
  onSubmit: (data) => void;
```

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 4]
```diff
@@ -21,19 +27,42 @@ const Root = styled.div`
...
   }, []);
 
   const formik = useFormik({
-    initialValues: { price: 0, name: " ", brand: " ", image: " " },
+    initialValues: {
+      price: productData?.price || 0,
+      name: productData?.name || "",
+      brand: productData?.brand || "",
+      image: productData?.image || "",
+    },
```
Comment (amitmarko) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx]:
```suggestion
    initialValues: initialValues || {
      price: 0,
      name: "",
      brand: "",
      image: "",
    },
```

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 5]
```diff
@@ -21,19 +27,42 @@ const Root = styled.div`
 const createProduct = (product: ICreateProduct) =>
   ProductService.createProduct(product);
 
-export const ProductForm = () => {
+export const ProductForm = ({
+  productData,
```
Comment (amitmarko) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx]:
```suggestion
  initialValues,
```

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 6]
```diff
@@ -91,10 +120,7 @@ export const ProductForm = () => {
           >
             {t("form.buttons.submit")}
           </Button>
-          <Button
-            onClick={() => navigation(URL.PRODUCTS)}
-            style={{ marginLeft: "55px" }}
-          >
+          <Button onClick={onCancel} style={{ marginLeft: "55px" }}>
```
Comment (amitmarko) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx]: Do not use `style` every, `sx` or `styled components`

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 1 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx]: Fixed

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 2 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx:12]: I pass it only in the Edit component,in the CreateProduct component i dont need to pass it(onCancel)

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 4 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx]: Fixed

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 5 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx]: Fixed

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 3 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx:13]: Fixed

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 6 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx]: Fixed

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 7]
```diff
@@ -6,34 +6,45 @@ import { Button } from "primereact/button";
...
   const { t } = useTranslation();
-  const navigation = useNavigate();
-  const { mutate, isLoading } = useMutation(createProduct, {
-    onSuccess: () => navigation(URL.PRODUCTS),
-  });
 
   const handleSubmit = useCallback((data: ICreateProduct) => {
-    mutate(data);
+    initialValues ? onSubmit(data) : onSubmit(data);
   }, []);
```
Comment (itairom-appsforce) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx]: remove

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 8]
```diff
@@ -6,34 +6,45 @@ import { Button } from "primereact/button";
...
   const formik = useFormik({
-    initialValues: { price: 0, name: " ", brand: " ", image: " " },
+    initialValues: initialValues || {
+      price: 0,
+      name: "",
+      brand: "",
+      image: "",
+    },
+
     onSubmit: handleSubmit,
```
Comment (itairom-appsforce) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx]:
```suggestion
    onSubmit,
```

File: src/Pages/CreateProduct/components/ProductForm/index.tsx [hunk 9]
```diff
@@ -6,34 +6,45 @@ import { Button } from "primereact/button";
...
-import { InputField } from "../../../../components/UI/Forms/InputField";
+import { InputField } from "../../../../Components/UI/Forms/InputField";
 import { validationYup } from "../../../../schema/validations/validationSchema";
 import { useTranslation } from "react-i18next";
 
+interface IProps {
+  initialValues?: IProduct;
+  onCancel?: () => void;
+  onDone?: () => void;
+  onSubmit: (data: ICreateProduct) => void;
```
Comment (itairom-appsforce) [inline src/Pages/CreateProduct/components/ProductForm/index.tsx]:
```suggestion
  onSubmit: (data: ICreateProduct | IProduct) => void;
```

File: src/schema/validations/initialValues.ts [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  brand: "",
+  image: "",
+};
+
+export const initialProductValues = (product: IProduct) => ({
+  price: product?.price,
+  name: product?.name,
+  brand: product?.brand,
+  image: product?.image,
+});
```
Comment (itairom-appsforce) [inline src/schema/validations/initialValues.ts]: delete file

File: src/ui-kit/modal/EditModal/index.tsx [hunk 11]
```diff
@@ -1,35 +1,50 @@
...
+import { ICreateProduct, IProduct } from "models";
+import { ProductService } from "../../../services/product-service";
+import { useMutation } from "react-query";
 
 interface IProps {
   visible: boolean;
   onCancel: () => void;
+  productId: string;
+  productData: IProduct;
+  closeModal: () => void;
```
Comment (itairom-appsforce) [inline src/ui-kit/modal/EditModal/index.tsx]: remove, its unused

File: src/schema/validations/initialValues.ts (hunk 10 above)
Comment (maximFeldman1) [reply src/schema/validations/initialValues.ts]: Fixed

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 9 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx]: Fixed

File: src/ui-kit/modal/EditModal/index.tsx (hunk 11 above)
Comment (maximFeldman1) [reply src/ui-kit/modal/EditModal/index.tsx]: FIxed

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 8 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx]: Fixed

File: src/Pages/CreateProduct/components/ProductForm/index.tsx (hunk 7 above)
Comment (maximFeldman1) [reply src/Pages/CreateProduct/components/ProductForm/index.tsx]: Fixed

**Pull request merged**